packages/buckets/src/api/index.ts

```diff
@@ -542,42 +578,63 @@ export async function bucketsPushPath(
             },
             root: event.root?.path ?? '',
           }
-          resolve(res)
+          return resolve(res)
         } else if (opts?.progress) {
           opts.progress(event?.bytes)
         }
       } else {
-        reject(new Error('Invalid reply'))
+        return reject(new Error('Invalid reply'))
       }
     })
     client.onEnd((code, msg) => {
       if (code !== grpc.Code.OK) {
         const message = msg ? msg : code.toString()
-        reject(new Error(message))
+        return reject(new Error(message))
```

Comment:
These returns avoid having to deal with catching already closed connections later on down the line. Once we close a connection, we're never going to use it again, so we can just return here.